src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3264,10 +3264,11 @@ private BoundExpression BindAsOperator(BinaryExpressionSyntax node, DiagnosticBa
             if (operand.IsLiteralDefault())
             {
                 var defaultLiteral = (BoundDefaultExpression)operand;
+                Debug.Assert((object)defaultLiteral.TargetType == null);
```

Comment:
>Debug.Assert((object)defaultLiteral.TargetType == null); [](start = 16, length = 56)

Why bother asserting this instead of simply using ```defaultLiteral.TargetType``` in the constructor below. That would be correct no matter what its value is.

Reply:
A default *literal* by definition does not have an explicit target type specified in source.  This is just asserting that fact.

---
In reply to: [274638771](https://github.com/dotnet/roslyn/pull/34889#discussion_r274638771) [](ancestors = 274638771)